### PR TITLE
Metadata implementation

### PR DESCRIPTION
**Pull request by nickhsine**
[KanbanFlow Card](https://kanbanflow.com/t/PjF4nyon)
[Zeplin Mockup](zpl.io/VKPn9vX)

### CODE REVIEW

File: src/components/article-page.js [hunk 1]
```diff
@@ -19,6 +20,14 @@ export default class Article extends PureComponent {
       <div>
         <LeadingBlock />
         <Body content={_.get(post, 'content.api_data')} />
+        <Metadata
+          categories={post.categories}
+          designers={post.designers}
+          photographers={post.photographers}
+          tags={post.tags}
+          themeColor="#fb9cec"
+          writers={post.writters}
```
Comment (YuCJ) [inline src/components/article-page.js]: [TODO] We need to add `engineers` and `extendLine` if the components are used for general purpose

File: src/components/metadata/index.js [hunk 2]
```diff
@@ -0,0 +1,248 @@
...
+        sort_order: PropTypes.number.isRequired,
+      })
+    ),
+    tags: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.string.isRequired,
+        name: PropTypes.string.isRequired,
+      })
+    ),
+    themeColor: PropTypes.string.isRequired,
```
Comment (YuCJ) [inline src/components/metadata/index.js]: I suggest to use the `ThemeProvider` of `styled-components` to store all theme related variables (which will be provided by the consumer of this package), then we can use these variables directly in any styled component at any nesting level and do not need to pass them via props.

File: src/components/metadata/index.js [hunk 3]
```diff
@@ -0,0 +1,248 @@
...
+
+    return (
+      <AuthorFlexBox>
+        <NoShrinkFlexItem>
+          <AuthorJobTitle fontSize="14" color="#808080" marginLeft="5">
+            {label}
+          </AuthorJobTitle>
+        </NoShrinkFlexItem>
+        <NoShrinkFlexItem>
+          <AngledSeparationLine color={themeColor} />
```
Comment (YuCJ) [inline src/components/metadata/index.js]: According to the mockup at zepline, the line seems should be placed behind the job title

**Review by YuCJ — COMMENTED** [review]
LGTM 👍

File: src/components/metadata/index.js (hunk 2 above)
Comment (YuCJ) [reply src/components/metadata/index.js]: See https://github.com/twreporter/react-article-components/pull/3/files

File: src/components/article-page.js (hunk 1 above)
Comment (nickhsine) [reply src/components/article-page.js]: Sure,  will add `engineers` first.
And will add `extend_byline` after mockup ready.

File: src/components/metadata/index.js (hunk 3 above)
Comment (nickhsine) [reply src/components/metadata/index.js]: Yeah, and it placed after job title already.

**Pull request merged**